### PR TITLE
Add TIF features to CLI to view filtered book and open orders

### PR DESCRIPTION
**Pull request by jarry-xiao**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -33,7 +33,8 @@ spl-token = { version = "3.2.0", features = ["no-entrypoint"] }
 serde = { version = "1.0", features = ["derive"] }
 serde_json = "1.0"
 spl-associated-token-account = { version = "1.1.1", features = [ "no-entrypoint" ] } 
-phoenix-v1 = "0.1.4"
-phoenix-sdk = { version = "0.1.0" }
+phoenix-v1 = { git = "https://github.com/Ellipsis-Labs/phoenix-v1", rev = "60aaa21" } 
+phoenix-sdk = { git = "https://github.com/Ellipsis-Labs/phoenix-sdk", rev = "cc2366d" }
```
Comment (jarry-xiao) [inline Cargo.toml:37]: NOTE: Change before merge

File: src/lib/helpers/market_helpers.rs [hunk 2]
```diff
@@ -115,15 +117,39 @@ pub async fn get_book_levels(
...
+        .value;
+
+    let market_account_data = market_and_clock
+        .remove(0)
+        .ok_or(anyhow::Error::msg("Market account not found"))?
+        .data;
+
+    let clock_account_data = market_and_clock
+        .remove(0)
+        .ok_or(anyhow::Error::msg("Market account not found"))?
```
Comment (eugene-chen) [inline src/lib/helpers/market_helpers.rs:135]: fix comment

File: src/lib/processor/process_get_open_orders.rs [hunk 3]
```diff
@@ -13,15 +17,34 @@ pub async fn process_get_open_orders(
...
+        .value;
+
+    let market_account_data = market_and_clock
+        .remove(0)
+        .ok_or(anyhow::Error::msg("Market account not found"))?
+        .data;
+
+    let clock_account_data = market_and_clock
+        .remove(0)
+        .ok_or(anyhow::Error::msg("Market account not found"))?
```
Comment (eugene-chen) [inline src/lib/processor/process_get_open_orders.rs:36]: and here

File: src/lib/processor/process_get_open_orders.rs [hunk 4]
```diff
@@ -54,6 +80,24 @@ pub async fn process_get_open_orders(
                     order.num_base_lots.as_u64() as f64 * sdk.base_lots_to_base_units_multiplier(),
                     size_precision
                 ),
+                format!(
+                    "{}",
+                    if order.last_valid_slot >= clock.slot {
+                        (1 + order.last_valid_slot - clock.slot).to_string()
+                    } else {
+                        "∞".to_string()
```
Comment (eugene-chen) [inline src/lib/processor/process_get_open_orders.rs:88]: does this need to be inside its own `format!`? can you just pass the raw string value to the line 70 `format!`?

File: src/lib/processor/process_get_open_orders.rs [hunk 5]
```diff
@@ -82,6 +129,24 @@ pub async fn process_get_open_orders(
                     order.num_base_lots.as_u64() as f64 * sdk.base_lots_to_base_units_multiplier(),
                     size_precision,
                 ),
+                format!(
```
Comment (eugene-chen) [inline src/lib/processor/process_get_open_orders.rs:132]: pull all this stuff into its own function to avoid duplicated code

**Review by throwbackjams — APPROVED** [review]
Looks good to me. One small nit is that we can put the get market data and clock into a helper function but if we don't expect additional future duplication, then probably not worth it.

**Pull request merged**